node_tree.go

```diff
@@ -258,15 +267,32 @@ func (t *NodeTree) Get(url string) (ok bool, n *Node, err error) {
 	return false, &Node{}, nil
 }
 
-// FullTextSearch uses a prebuild search index to perform a search
+// FullTextSearch uses a prebuilt search index to perform a search
 // over all possible attributes of each node.
 func (t *NodeTree) FullTextSearch(query string) ([]*Node, int, time.Duration) {
 	start := time.Now()
 
-	t.RLock()
-	defer t.RUnlock()
+	textQuery := bleve.NewMatchQuery(query)
+	textQuery.SetFuzziness(2)
+	prefixQuery := bleve.NewPrefixQuery(query)
+	termQuery := bleve.NewTermQuery(query)
+	disjunctionQuery := bleve.NewDisjunctionQuery(prefixQuery, textQuery, termQuery)
```

Comment:
I didn't know mixing query types is possible, can you explain why it is beneficial to use all three?

Reply:
At least with the internal app, not sure if I can mention it publicly, that forms the basis of this work, we dealt with results where certain things that should have matched with a raw match query, did not.

For example, `Farben` being an article we wanted to match, we would type

| Input        | True positive|
| ------------- |:-------------:| 
| f | true|
| fa | true|
|far| false|
|farb |false|
|farbe| true|
|farben|true|

What is used by bleve for fuzzy matching under the hood, Levinstein distances weren't enough  weren't able to match this on its own. Especially for just a few characters typed.

The termQuery, of the three, is probably the least useful, as it assumes that someone knows the entirety of exactly what they're looking for. It should probably be removed.

> A term query is the simplest possible query. It performs an exact match in the index for the provided term.
> 
> Most of the time users should use a Match Query instead.